app/models/org.rb

```diff
@@ -178,20 +220,25 @@ def grant_api!(token_permission_type)
   end
 
   private
-    ##
-    # checks size of logo and resizes if necessary
-    #
-    def resize_image
-      unless logo.nil?
-        if logo.height != 100
-          self.logo = logo.thumb('x100')  # resize height and maintain aspect ratio
-        end
-      end
-    end
 
-    # creates a dfefault Guidance Group on create on the Org
-    def create_guidance_group
-      GuidanceGroup.create(name: self.abbreviation? ? self.abbreviation : self.name , org_id: self.id)
+  ##
+  # checks size of logo and resizes if necessary
+  #
+  def resize_image
+    unless logo.nil?
+      if logo.height != 100
+        self.logo = logo.thumb('x100')  # resize height and maintain aspect ratio
+      end
     end
+  end
 
+  # creates a dfefault Guidance Group on create on the Org
+  def create_guidance_group
+    GuidanceGroup.create!({
+      name: abbreviation? ? self.abbreviation : self.name ,
```

Comment:
@Bodacious what are your thoughts on models creating other models like this? Is it good practice? The other option I suppose would be to have this method on the `guidance_group` itself if not just a part of its `initialize` method.

We've gone back and forth on this project. It originally concerned me because it wasn't always apparent (due to method names) that a record was being created. For example we at one point had `copy = Template.deep_copy` which would make me expect that a copy of the model and its dependencies had been created but not yet persisted when in fact `create!` had been called for all of the objects. 

The name for this one is much clearer than deep_copy so maybe its ok. Should we add a '!' to the end of it though?

Reply:
It's not the worst approach. But I think a service object would create a better de-coupling, and make it clearer what is happening.